File: tests/benchmark/optimum_fibonacci.pl
#!/usr/bin/perl

sub fibo
{
	my ($x) = @_;

	if ($x<=1) {
		return $x;
	}
	
	$f=1;
	$fp=1;
	$i=1;

	while ($i<$x)
	{
		$temp = $f;
		$f += $fp;
		$fp = $temp;
		$i+=1;
	}
	return $f;
}

print fibo(@ARGV[0]+0)." ";

File: tests/benchmark/optimum_fibonacci.rb
#!/usr/bin/ruby
def fibo(x)
	if x<=1
		return x
	end
	f=1
	fp=1
	i=1

	while i<x
		temp = f
		f += fp
		fp = temp
		i+=1
	end
	return f
end

puts fibo(ARGV[0].to_i)